astropy/cosmology/flrw/base.py

```diff
@@ -1678,9 +1645,9 @@ def __post_init__(self):
         object.__setattr__(self, "_Ode0", 0)
         super().__post_init__()
         # Do some twiddling after the fact to get flatness
-        object.__setattr__(self, "_Ok0", 0.0)
-        object.__setattr__(
-            self, "_Ode0", 1.0 - (self._Om0 + self._Ogamma0 + self._Onu0 + self._Ok0)
+        self.__dict__["Ok0"] = 0.0
+        object.__setattr__(  # managed by a Parameter
+            self, "_Ode0", 1.0 - (self.Om0 + self.Ogamma0 + self.Onu0 + self.Ok0)
         )
 
     @lazyproperty
```

Comment:
This can probably a `cached_property` too, no? (Obviously, doesn't really matter, but since `Cosmology` instances cannot be changed, the `__set__` and `__del__` methods of `lazyproperty` are not used anyway)